plugins/reach/lib/model/EntryVendorTaskPeer.php

```diff
@@ -34,35 +34,66 @@ public static function retrieveEntryIdAndCatalogItemIdAndEntryVersion($entryId,
 		$c->add(EntryVendorTaskPeer::VERSION, $version);
 		return EntryVendorTaskPeer::doSelect($c);
 	}
-	
-	public static function retrieveActiveTasks($entryId, $catalogItemId, $partnerId, $version, $statuses = array())
+
+    public static function retrieveTasksByStatus ($entryId, $catalogItemId, $partnerId, $version, array $statuses)
+    {
+        $c = new Criteria();
+        $c->add(EntryVendorTaskPeer::ENTRY_ID, $entryId);
+        $c->add(EntryVendorTaskPeer::CATALOG_ITEM_ID, $catalogItemId);
+        $c->add(EntryVendorTaskPeer::PARTNER_ID, $partnerId);
+        $c->add(EntryVendorTaskPeer::STATUS, $statuses, Criteria::IN);
+        if ($version)
+        {
+            $c->add(EntryVendorTaskPeer::VERSION, $version);
+        }
+
+        $c->addDescendingOrderByColumn(EntryVendorTaskPeer::VERSION);
+        $c->addDescendingOrderByColumn(EntryVendorTaskPeer::ID);
+
+        return EntryVendorTaskPeer::doSelect($c);
+    }
+
+	public static function retrieveOneTaskByStatus ($entryId, $catalogItemId, $partnerId, $version, array $statuses)
+    {
```

Comment:
duplicate code, either use a function that build the criteria and reused in this 2 functions, or in this funvtion - call retrieveTasksByStatus and just return the head of the array.